### ISSUE TITLE
the kernel will be interrupted when execute in jupyter lab

### ISSUE THREAD

**Issue by hgzhangs** [GeoDaCenter/pygeoda#1]
reassign a instance to geoda object after calculate  weight will interrupt the kernel
1. gda = geopandas_to_geoda(gdf)
2. w = knn(gda, **kwargs)
3. gda = new_obj

**Comment by lixun910** [comment]
Thanks, @hgzhangs  Could you provide more details? Here is what I have on my side: python 3.8 / Mac OSX 10.15.6

<img width="1152" alt="Screen Shot 2020-11-03 at 11 29 20 AM" src="https://user-images.githubusercontent.com/2423887/98026176-f5889b00-1dc7-11eb-81e0-9bcef35809d6.png">

**Comment by hgzhangs** [comment]
this code run on Windows 10 20H2/Python 3.6
![image](https://user-images.githubusercontent.com/11520371/98057040-4954bc00-1e7c-11eb-91e6-b22b828d39ab.png)
![image](https://user-images.githubusercontent.com/11520371/98057006-35a95580-1e7c-11eb-85fb-afd59a0b92c7.png)

**Comment by lixun910** [comment]
@hgzhangs Thanks!  Just want to be clear: does it crash with the following lines:
```Python
new_gdf = gdp.read_file("......period_data.shp")
yearly_stations_gda = pygeoda.geopandas_to_geoda(new_gdf)
```
Thanks!

**Comment by hgzhangs** [comment]
yes, the kernel will crash in those line
> @hgzhangs Thanks! Just want to be clear: does it crash with the following lines:
> 
> ```python
> new_gdf = gdp.read_file("......period_data.shp")
> yearly_stations_gda = pygeoda.geopandas_to_geoda(new_gdf)
> ```
> 
> Thanks!

**Comment by lixun910** [comment]
Thanks @hgzhangs !   Does read the shp file using pygeoda.open() crash as well?

```python
yearly_station_gda = pygeoda.open('....period_data.shp')
```
Do you know if there is null/empty  geometry in the dataset? You can use `GeoSeries.is_empty` to check. see: https://geopandas.org/missing_empty.html

Thanks!

**Comment by hgzhangs** [comment]
@lixun910 Thanks, 
pygeoda.open() also interrupt the kernel as well,
![image](https://user-images.githubusercontent.com/11520371/98059968-7193e900-1e83-11eb-91c0-ec356cfb39db.png)
and I have checked my dataset, may there is not null geometry in this dataset.
![image](https://user-images.githubusercontent.com/11520371/98060458-9472cd00-1e84-11eb-8f89-34d0469cf3ce.png)
Thanks!!

**Comment by lixun910** [comment]
Got it. Is it possible to share you data with me so I can check? I will keep your data confidential. Thanks! My email: xunli at UChicago.edu

Xun


Xun Li, Ph.D.

Assistant Director of Data Science
Center for Spatial Data Science
University of Chicago

**Comment by lixun910** [comment]
fixed in v0.0.8
